servicereportpkg/validate/plugins/spyre.py

```diff
@@ -248,14 +250,26 @@ def check_vfio_access_permission(self):
 
         if not os.path.isdir(vfio_dir):
             self.log.error("No %s directory", vfio_dir)
+            perm_check.set_status(False)
+            return perm_check
+
+        group_name = 'sentient'
+        try:
+            gid = grp.getgrnam(group_name).gr_gid
+        except Exception as e:
+            self.log.error("Failed to get groupid of group: %s", group_name)
+            perm_check.set_status(False)
             return perm_check
 
         for name in os.listdir(vfio_dir):
             full_path = vfio_dir + name
             try:
+                ret = True
+                if os.stat(full_path).st_gid != gid:
+                    ret = False
```

Comment:
This checks all files under `vfio_dir`, whereas the check immediately below
`if stat.S_ISCHR(mode):` looks only for character device files inside
`vfio_dir`.

May I know why we are checking files other than character devices?

Even if their group ID does not match the expected value, those files will not be
added to the per_check object.

We can address this in one of two ways:
1. If there is a need to check all files under `vfio_dir`, remove the
character device check so that both the group ID and ownership checks are
performed for all files.

2. Else keep the group ID check within the `if stat.S_ISCHR(mode)`: condition.